src/test/java/unstable/RepositoryPickerTest.java

```diff
@@ -0,0 +1,211 @@
+        doubleClick(userInputField);
+        type("dummmy");
+        assertEquals(1, suggestedRepositoryList.getChildren().size());
+        assertSelectedPickerRepositoryNode("dummmy", suggestedRepositoryList.getChildren().get(0));
+        push(KeyCode.ESCAPE);
+
+        // we check if repo's id with white spaces are handled correctly
+        traverseMenu("Repos", "Show Repository Picker");
+        PlatformEx.waitOnFxThread();
+        userInputField = findOrWaitFor(IdGenerator.getRepositoryPickerTextFieldReference());
+        doubleClick(userInputField);
+        doubleClick();
+        type(" dummy4 / dummy4 ");
+        push(KeyCode.ENTER);
+        traverseMenu("Repos", "Show Repository Picker");
+        PlatformEx.waitOnFxThread();
+        suggestedRepositoryList = findOrWaitFor(IdGenerator.getRepositoryPickerSuggestedRepoListReference());
+        assertEquals(4, suggestedRepositoryList.getChildren().size());
+        assertEquals("dummy4/dummy4", primaryRepo);
+        push(KeyCode.ESCAPE);
+
+        // we check if deleting used repo does not remove it
+        traverseMenu("Repos", "Remove", "dummy4/dummy4 [in use, not removable]"); // first used repo
+        push(KeyCode.ENTER);
+        PlatformEx.waitOnFxThread();
+        traverseMenu("Repos", "Show Repository Picker");
+        PlatformEx.waitOnFxThread();
+        suggestedRepositoryList = findOrWaitFor(IdGenerator.getRepositoryPickerSuggestedRepoListReference());
+        assertEquals(4, suggestedRepositoryList.getChildren().size());
+        push(KeyCode.ESCAPE);
+
+        // we check if delete repo works
+        traverseMenu("Repos", "Remove", "dummy/dummy");
+        push(KeyCode.ENTER);
+        PlatformEx.waitOnFxThread();
+        traverseMenu("Repos", "Show Repository Picker");
+        PlatformEx.waitOnFxThread();
+        suggestedRepositoryList = findOrWaitFor(IdGenerator.getRepositoryPickerSuggestedRepoListReference());
+        assertEquals(3, suggestedRepositoryList.getChildren().size());
+        push(KeyCode.ESCAPE);
+
+        // we check again if deleting used repo does not remove it
+        traverseMenu("Repos", "Remove", "dummy2/dummy2 [in use, not removable]"); // second used repo
+        push(KeyCode.ENTER);
+        PlatformEx.waitOnFxThread();
+        traverseMenu("Repos", "Show Repository Picker");
+        PlatformEx.waitOnFxThread();
+        suggestedRepositoryList = findOrWaitFor(IdGenerator.getRepositoryPickerSuggestedRepoListReference());
+        assertEquals(3, suggestedRepositoryList.getChildren().size());
+        push(KeyCode.ESCAPE);
+
+        // exit program
+        traverseMenu("File", "Quit");
+        push(KeyCode.ENTER);
+
+        // testing that the correct repo was saved in the json
+        // check if the test JSON is still there...
+        if (!(testConfig.exists() && testConfig.isFile())) {
+            fail();
+        }
```

Comment:
It's from `RepositorySelectorTest`. That seems better, will change it to `assertTrue`.